### PR TITLE
I367

### LINKED ISSUE #367: allow users to change their password after authentication
Once users have successfully logged in once, should allow them to change their password

### PR DESCRIPTION
**Pull request by adamrp**
Fixes #367. Adds change password functionality.  Specific questions to consider before possible merge:
1) icon for side menu okay?  Taken from: http://www.iconfinder.com/icondetails/80968/64/key_icon
2) Are there any policies we want to enforce about the password creation (right now, it accepts anything)?

### CODE REVIEW

**Comment by ElDeveloper** [comment]
Some notes:
- Old and new passwords are visible as you type.
- Submit button doesn't seem to be working.
- Could we have some javascript voodo magic that checks that the new passwords match? Perhaps with a checkmark at the side of the text entry?
- Add some instructions/information to the Change Password section maybe using the jQuery thing with the question mark.

Would it be reasonable to send e-mail confirmation about password change?

**Comment by adamrp** [comment]
- Good point on password visibility.  I'll work on that.  I guess they should all be hidden.
- The submit button works for me on VB and on webdev.  Do you have any javascript console errors or anything?
- I do check to make sure the new passwords (using PSP not javascript).  If they do not match, a message is displayed at the top of the page.
- Not sure if additional instructions are necessary (e.g., popup text in jquery boxes), but I don't have strong feelings about.  Others' thoughts?
- I was wondering about sending an email out.  I don't think it would be hard to do.

**Comment by ElDeveloper** [comment]
We can take a look at my password username pairs and see if it's something related to that.

Just checked and I do have a console error:

13fusebox.psp:116ReferenceError: Can't find variable: validatePasswordReset

> I do check to make sure the new passwords (using PSP not javascript). If they do not match, a message is displayed at the top of the page.
> I was thinking on having something that checks as you type, though that might be out of the scope? The idea behind this is to try and make it such that you don't click submit or you can't click submit unless the passwords match as it is something that can be checked before submission.

Sending the e-mail shouldn't be hard.

**Comment by adamrp** [comment]
Hmm.  Are you testing in VB, or on webdev?  If on VB, remember you will have to open the JS file from within VB to kick it.  If on webdev, I'll come have a look with you.

**Comment by ElDeveloper** [comment]
No, testing it in webdev.

**Comment by douginator2000** [comment]
I've noticed at times the site(s) will not re-load, but if I open a
different browser (e.g. I'm using Chrome then switch to Safari) it works
immediately. This could be some odd issue with the site... or something
else. I'm not really sure - it's just a behavior I have noticed.

On Mon, Jul 1, 2013 at 1:52 PM, Yoshiki Vázquez Baeza <
notifications@github.com> wrote:

> No, testing it in webdev.
> 
> El 01/07/2013, a las 13:50, adamrp notifications@github.com escribió:
> 
> > Hmm. Are you testing in VB, or on webdev? If on VB, remember you will
> > have to open the JS file from within VB to kick it. If on webdev, I'll come
> > have a look with you.
> > 
> > —
> > Reply to this email directly or view it on GitHub.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/qiime/qiime_web_app/pull/454#issuecomment-20305848
> .

## 

Doug Wendel
Professional Research Associate
University of Colorado at Boulder
Department of Chemistry and Biochemistry, Knight Lab
303-641-5442
wendel@colorado.edu

**Comment by adamrp** [comment]
Okay, I have implemented all requested changes so far except for checking that the new password and the confirm password entries match in real time using JS (imho don't need it, but if i hear other requests for it, I can do that).

File: www/americangut/change_password.psp [hunk 1]
```diff
@@ -0,0 +1,92 @@
+<%
```
Comment (ElDeveloper) [inline www/americangut/change_password.psp:1]: Would be good to add the dosctrings at the top of the file, i. e. **author**, **email**, etc.

It's not necessarily bad but there is only one comment :warning:

**Comment by ElDeveloper** [comment]
Yeah, the JS thing doesn't sound like something that we need.

Just took a look and it's all looking great!

Perhaps the only two open points in this are:
    - Do we want password policies? (do we even care if participants have an insecure password).
    - What to do about the icon credit attribution?

If someone else can take a quick look at this, we should be ready to merge this!

**Comment by douginator2000** [comment]
Nice, I will look at this tomorrow am.

On Monday, July 1, 2013, Yoshiki Vázquez Baeza wrote:

> Yeah, the JS thing doesn't sound like something that we need.
> 
> Just took a look and it's all looking great!
> 
> Perhaps the only two open points in this are:
> - Do we want password policies? (do we even care if participants have an
>   insecure password).
> - What to do about the icon credit attribution?
> 
> If someone else can take a quick look at this, we should be ready to merge
> this!
> 
> El 01/07/2013, a las 16:43, adamrp <notifications@github.com<javascript:_e({}, 'cvml', 'notifications@github.com');>>
> escribió:
> 
> > Okay, I have implemented all requested changes so far except for
> > checking that the new password and the confirm password entries match in
> > real time using JS (imho don't need it, but if i hear other requests for
> > it, I can do that).
> > 
> > —
> > Reply to this email directly or view it on GitHub.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/qiime/qiime_web_app/pull/454#issuecomment-20316419
> .

## 

Doug Wendel
Professional Research Associate
University of Colorado at Boulder
Department of Chemistry and Biochemistry, Knight Lab
303-641-5442
wendel@colorado.edu

**Comment by ElDeveloper** [comment]
Cool thanks!

**Comment by douginator2000** [comment]
Tested this morning - looks good to me.

**Pull request merged**